### PR TITLE
Fix:5175 Upgraded Minimum SDK Version, Compile and Target SDK Version

### LINKED ISSUE #5175: Upgrade Minimum SDK Version, Compile and Target SDK Version
### What is the user problem or growth opportunity you want to see solved?

As part of https://github.com/commons-app/apps-android-commons/issues/4994 and based on https://github.com/commons-app/apps-android-commons/issues/4994#issuecomment-1442125601, before migrating from Mapbox to MapLibre, we require certain upgrades in dependencies in our application. This enhancement is to upgrade the Minimum SDK Version, Compile and Target SDK Versions of the application.

minSdkVersion: 19 -> 21
compileSdkVersion: 28 -> 31
targetSdkVersion: 28 -> 31 

### How do you know that this problem exists today? Why is this important?

_No response_

### Who will benefit from it?

_No response_

### Anything else you would like to add?

_No response_

### PR DESCRIPTION
**Pull request by kartikaykaushik14**
**Description (required)**

Fixes #5175 

What changes did you make and why?

Upgraded Minimum SDK Version, Compile and Target SDK Version

**Tests performed (required)**

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

**Screenshots (for UI changes only)**

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

**Review by nicolas-raoul — COMMENTED** [review]
Thanks! Are the unit tests running fine?

**Comment by kartikaykaushik14** [comment]
> Thanks! Are the unit tests running fine?

I tried the changes on top of the fix I made for #5164 and the unit tests are successful.

**Comment by nicolas-raoul** [comment]
Wonderful! Would you mind pulling the https://github.com/commons-app/apps-android-commons/pull/5164 branch into here? Sorry for the inconvenience!

**Comment by kartikaykaushik14** [comment]
The tests were passing but there were issues with the app startup and I'm working on them. I will update you as soon as I resolve them.

**Comment by kartikaykaushik14** [comment]
I was able to resolve the issues and cherry-picked the commits from #5164 branch here. Could you please rigorously test the PR for different flows?

**Comment by nicolas-raoul** [comment]
Bad luck, I am getting this:
```
An exception has occurred in the compiler (1.8.0_221)
java.lang.AssertionError: annotationType(): unrecognized Attribute name MODULE (class com.sun.tools.javac.util.UnsharedNameTable$NameImpl)
        at com.sun.tools.javac.util.Assert.error(Assert.java:133)
```
@Ayan-10 @sivaraam  would you mind testing this PR for most app flows? Thanks!

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageLoader.kt [hunk 1]
```diff
@@ -138,6 +138,18 @@ class ImageLoader @Inject constructor(
                             is Result.TRUE -> {
                                 mapResult[imageSHA1] = Result.TRUE
                             }
+                            is Result.ERROR -> {
```
Comment (Ayan-10) [inline app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageLoader.kt:141]: @nicolas-raoul The app is running without any crashes, all test cases are also getting passed, just some addition of tests for these new lines would be better.

**Comment by sivaraam** [comment]
> would you mind testing this PR for most app flows? Thanks!

I tested basic flows like viewing contributions, explore, peer review, nearby. They all seem to be working fine to me. Let me know in case anything else needs to be checked in specific.

**Comment by kartikaykaushik14** [comment]
@nicolas-raoul Could you please let me know how we're planning to proceed so I can work accordingly for the F-Droid issue?

**Comment by nicolas-raoul** [comment]
@kartikaykaushik14 Apologies for putting you in this situation. The last 10 merged pull requests were all merged by me, so it is probable that if I am not able to run the app anymore, the app's development and Google Summer of Code will both stall.

Would you mind rebasing? Never know, slightly different code might prevent my compiler from crashing. Thanks a lot for your understanding and patience!

File: gradle.properties [hunk 2]
```diff
@@ -13,23 +13,24 @@
...
 MULTIDEX_VERSION=2.0.1
-OKHTTP_VERSION=3.12.1
+OKHTTP_VERSION=4.10.0
 
 systemProp.http.proxyPort=0
 systemProp.http.proxyHost=
 android.useAndroidX=true
 android.enableJetifier=true
 android.enableR8=false
+android.jetifier.blacklist=bcprov-jdk15on
```
Comment (nicolas-raoul) [inline gradle.properties:36]: `blacklist` is deprecated, better use `ignorelist`.

**Review by nicolas-raoul — APPROVED** [review]
I rebased and switched to Zulu JDK, and finally it worked. :-)

**Pull request merged**